### PR TITLE
Add fm docs

### PR DESCRIPTION
**Pull request by lars-petter-hauge**
**Task**
Add documentation regarding forward models that comes with libres.

### CODE REVIEW

File: docs/rst/manual/forward_model/index.rst [hunk 1]
```diff
@@ -11,45 +11,229 @@ modelling tasks like e.g. relperm interpolation or water saturation, or to
...
+default behaviour. The jobs are typically quite strict, i.e. if you
+try to remove a directory with the :code:`DELETE_FILE` job, or visa
+verse, nothing will happen. The actions performed will typically be
+logged to the :code:`xxx.stdout` files:
+
+.. code-block:: bash
+
+    Copying file 'file.txt' to 'target_backup.txt'
+
+And error messages will go to the :code:`xxx.stderr` files. None of
```
Comment (markusdregi) [inline docs/rst/manual/forward_model/index.rst]: This is not true, a non-zero return value will be picked up. This must be quite outdated!!!

File: docs/rst/manual/forward_model/index.rst [hunk 2]
```diff
@@ -11,45 +11,229 @@ modelling tasks like e.g. relperm interpolation or water saturation, or to
...
 
 Reservoir modelling: RMS
 ........................
 
 File system utilities
 .....................
 
+There are many small jobs for copying and moving files and
+directories. The jobs are just thin wrappers around the corresponding
+unix commands, i.e. :code:`/bin/cp` and :code:`/bin/ln`, but we have
```
Comment (markusdregi) [inline docs/rst/manual/forward_model/index.rst]: This is also not true as they are implemented in Python. But it should not matter, as it is an implementation detail...

**Comment by markusdregi** [comment]
The target file section must be removed, otherwise this seems to be correct. A welcome initiative 👍

File: docs/rst/manual/forward_model/index.rst (hunk 1 above)
Comment (lars-petter-hauge) [reply docs/rst/manual/forward_model/index.rst]: Probably, we’ll have to go through this a few times I guess. Thanks for notifying about this one though, I’ll get it updated! :)

Comment (lars-petter-hauge) [reply docs/rst/manual/forward_model/index.rst]: Given that you're providing a non-existing file, the forward model will be considered successful, and the following will be displayed in the `stderror` file of the job:

`File: '/path/to/file/delete_me' not exist - delete ignored`

Trying to delete a directory with the `DELETE_FILE` job, the forward model will fail with the typical `[000:0000-0000] FAILED COMPLETELY.` and traceback from `stderror` then displays:

```
Traceback (most recent call last):
  File "/work/install/ert/share/ert/forward-models/shell/script/delete_file", line 8, in <module>
    delete_file( file )
  File "/work/install/ert/lib/python2.7/site-packages/res/fm/shell/__init__.py", line 13, in delete_file
    Shell.deleteFile(filename)
  File "/work/install/ert/lib/python2.7/site-packages/res/fm/shell/shell.py", line 134, in deleteFile
    raise IOError("Entry:'%s' is not a regular file" % filename)
IOError: Entry:'delete_me' is not a regular file
```
I suppose that failing a job -> fails the forward model, is sensible. Is it just as sensible ignoring the delete if the file does not exist? What if the user has misspelled the name?

Comment (markusdregi) [reply docs/rst/manual/forward_model/index.rst]: I agree that this should be discussed, but we should then agree on consistent behaviour for all of the shell commands. Also, changes like this should be done with care...

Let's be happy for now if the documentation adheres to the current behaviour ;)

**Comment by lars-petter-hauge** [comment]
@markusdregi have you got any more comments? Note that the commit concerning custom jobs is self-written https://github.com/equinor/ert/pull/289/commits/c26f5192627d2deac8329a68577e117d52990f79 But I've double checked the behaviour.

Otherwise this can be merged.

**Comment by markusdregi** [comment]
@lars-petter-hauge Ship it! Could you backport it to version-2.4 as well?

**Pull request merged**